components/script/script_runtime.rs

```diff
@@ -826,7 +832,46 @@ impl StreamConsumer {
 }
 
 #[allow(unsafe_code)]
-unsafe extern "C" fn report_stream_error_callback(_cx: *mut JSContext, error_code: usize) {
+unsafe extern "C" fn ConsumeStreamCallback(
+    _cx: *mut RawJSContext,
+    obj: HandleObject,
+    _mimeType: MimeType,
+    _consumer: *mut JSStreamConsumer,
+) {
+    let cx = JSContext::from_ptr(_cx);
+    let global = GlobalScope::from_context(*cx);
+    let returnValue = Promise::new(&global);
+    let mut handleobj = UndefinedValue();
+    GetProxyPrivate(*obj, &mut handleobj);
+    rooted!(in(*cx) let target = handleobj.to_object());
+    if let Ok(unwrappedSource) = root_from_handleobject::<Response>(target.handle(), *cx) {
+        let mimetype = unwrappedSource.Headers().extract_mime_type();
+        if std::str::from_utf8(&mimetype).unwrap() != "application/wasm" {
```

Comment:
Instead of using from_utf8, we can do: `if mimetype == b"application/wasm" {`

Reply:
Done